lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -165,7 +167,23 @@ try {
         Exit-Json $result
     }
 
-    $principal = New-ScheduledTaskPrincipal -UserId "$user" -LogonType ServiceAccount
+    # Handle RunAs/RunLevel options for the task
+
+    if (!$store_password) {
+        # Create a ScheduledTaskPrincipal for the task to run under
+        $principal = New-ScheduledTaskPrincipal -UserId $user -LogonType S4U -RunLevel $runlevel -Id Author
```

Comment:
Technically with the way this works, with a principal created for a service account set as S4U, it gets converted to ServiceAccount upon the task registering, and otherwise respects S4U for a typical user account.  That's what my testing shows.

Because of that, this acts as sort of a catch-all, rather than doing a specific check for service account names since it handles it anyway.